packages/bitcore-node/src/providers/chain-state/internal/internal.ts

```diff
@@ -466,9 +466,16 @@ export class InternalStateProvider implements CSP.IChainStateService {
     };
   }
 
-  async getDailyTransactions({ chain, network }: { chain: string; network: string }) {
-    const beforeBitcoin = new Date('2009-01-09T00:00:00.000Z');
-    const todayTruncatedUTC = new Date(new Date().toISOString().split('T')[0]);
+  async getDailyTransactions(params: CSP.DailyTransactionsParams) {
+    const { chain, network, startDate, endDate } = params;
+    const formatDate = (d: Date) => new Date(d.toISOString().split('T')[0]);
+    const todayTruncatedUTC = formatDate(new Date());
+    let oneMonth = new Date(todayTruncatedUTC);
+    oneMonth.setDate(todayTruncatedUTC.getDate() - 30);
+    oneMonth = formatDate(oneMonth);
+
+    const start = startDate ? new Date(startDate) : oneMonth;
```

Comment:
is request's `startDate` / `endDate` are invalid, "new Date(startDate)" will return "invalid Date"
that should be handled.